src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorageService.cs

```diff
@@ -19,6 +20,59 @@ internal partial class SQLitePersistentStorageService : AbstractPersistentStorag
 
         private readonly IPersistentStorageFaultInjector _faultInjectorOpt;
 
+        [DllImport("kernel32.dll")]
+        private static extern IntPtr LoadLibrary(string dllToLoad);
+        private static bool _initialized;
+        private static object _gate = new object();
```

Comment:
it might make a lot of sense to extract all this code related to laoding sqlite and put it in its own class.  The class would expose almost nothing except a bit saying if you can use sqlite or not.  You would never have to care or look for exceptions relating to asking the question in teh first place if you can use it.